framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/runtimes/AbstractLicenseOnlyDownloadExecutor.java

```diff
@@ -115,27 +118,28 @@ public WorkflowResponse executeDownload(DownloadSingleRuntimeRequest req) {
 	
 	private void initiateDownload(DownloadSingleRuntimeRequest req, DownloadRuntime dlrt, 
 			IRuntimeInstaller installer, File uniqueLoc, File downloads) {
-		String jobId = model.getJobManager().generateJobId();
 		String jobName = "Download runtime: " + dlrt.getId();
-		final IJob job = new SimpleJob(jobName, jobId);
-		model.getJobManager().addJob(job);
-		new Thread(jobName) {
-			public void run() {
+		
+		IStatusRunnableWithProgress task = new IStatusRunnableWithProgress() {
+			
+			@Override
+			public IStatus run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
```

Comment:
how would I as a user-programmer pass in my monitor? I would have expected ```#execute``` and ```#executeDownload``` to feature this parameter but it's not there.  What am I missing? Can you please explain?

Reply:
what's the reason for making ```#executeDownload``` public?